src/fairchem/core/datasets/base_dataset.py

```diff
@@ -34,8 +33,11 @@
 T_co = TypeVar("T_co", covariant=True)
 
 
-class DatasetMetadata(NamedTuple):
-    natoms: ArrayLike | None = None
+class DatasetMetadata:
```

Comment:
why even have this class if we can just dump random things into it? might as well just make it a dict?

Reply:
I think its better to keep it as a NamedTuple or a dataclass and fields that we decide are needed. This is a much more disciplined interface, so that we are not just dumping random things.

Reply:
We discussed this offline and agreed to follow up after this PR and figure out a better way to include Metadata inside of dataset object and make it fast enough to filter on.